### PR TITLE
Add benchmarks workflow

### PR DESCRIPTION
**Pull request by sushildeep**
https://jira.nuxeo.com/browse/DEVOPS-183

### CODE REVIEW

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    branches:
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
```
Comment (alxgomz) [inline .github/workflows/deploy-nuxeo.com-benchmarks.yml]: I guess this can be bumped

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    - name: Install Hugo
+      run: |
+        wget https://github.com/spf13/hugo/releases/download/v0.80.0/hugo_0.80.0_Linux-64bit.tar.gz -O /tmp/hugo.tgz
+        (cd /tmp/; tar xzvf hugo.tgz)
+
+    - name: Build Hugo Site
+      run: /tmp/hugo -t hyde
```
Comment (alxgomz) [inline .github/workflows/deploy-nuxeo.com-benchmarks.yml]: You might want to take alook at https://github.com/marketplace/actions/hugo-site that builds the site without installation requirements.

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    steps:
+    - name: Rsync to Server
+      run: |
+        rsync -Waxv --delete ${{ github.workspace }}/public/ root@www:/var/www/nuxeo.com/benchmarks
+
+    - name: Test Nginx Configuration
+      run: ssh root@www "nginx -t"
+
+    - name: Reload Nginx
+      run: ssh root@www "service nginx reload"
```
Comment (alxgomz) [inline .github/workflows/deploy-nuxeo.com-benchmarks.yml]: This step will require a deplo key and apporpriate server configuration.

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+name: Build and Deploy Nuxeo Benchmarks
+
+on:
+  push:
+    branches:
+      - main
```
Comment (alxgomz) [inline .github/workflows/deploy-nuxeo.com-benchmarks.yml]: there is no such branch

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml (hunk 3 above)
Comment (alxgomz) [reply .github/workflows/deploy-nuxeo.com-benchmarks.yml]: let's make sure we fix this before merging

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml (hunk 2 above)
Comment (alxgomz) [reply .github/workflows/deploy-nuxeo.com-benchmarks.yml]: don't like it?

**Review by alxgomz — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+jobs:
+  build:
+    runs-on: benchmarks
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Build Site
+      uses: chabad360/hugo-actions@master
```
Comment (alxgomz) [inline .github/workflows/deploy-nuxeo.com-benchmarks.yml]: please set`hugoVersion` & `args` params to match what was present before.

**Review by alxgomz — CHANGES_REQUESTED** [review]

**Comment by alxgomz** [comment]
I'm worried that this is deploying in prod without any prior test.
Were there any test worklow in the previous pipeline?

**Comment by alxgomz** [comment]
⬆️ @troger ?

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml (hunk 5 above)
Comment (sushildeep) [reply .github/workflows/deploy-nuxeo.com-benchmarks.yml]: sure

Comment (alxgomz) [reply .github/workflows/deploy-nuxeo.com-benchmarks.yml]: I forgot to mention it but do not use `master` as a ref. here you can use `v2` which is the latest. That will prevent you from being hit by non-backward compatible changes that may be introduced in master

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      with:
+        buildPath: 'public'
+        hugoVersion: 'extended_0.80.0'
+        args: hugo -t hyde
```
Comment (alxgomz) [inline .github/workflows/deploy-nuxeo.com-benchmarks.yml]: you just need the arguments, not the full command. Check the action. doc: https://github.com/chabad360/hugo-actions#-inputs

**Review by alxgomz — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml (hunk 5 above)
Comment (alxgomz) [reply .github/workflows/deploy-nuxeo.com-benchmarks.yml]: as mentionned by @gionn during the daily yesterday we should use SHA hashes and not tags or branch names anymore.

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        args: --gc --minify --cleanDestinationDir
+
+  deploy:
+    runs-on: benchmarks
+    needs: build
+
+    steps:
+      - name: Rsync to Server
+        run: |
+          rsync -Waxv --delete ${{ github.workspace }}/public/ root@www:/var/www/nuxeo.com/benchmarks
```
Comment (alxgomz) [inline .github/workflows/deploy-nuxeo.com-benchmarks.yml]: I checked on the target server and the files and the directory which holds them belong to `jenkins` (uid 1001).
That means there is no need to rsync as `root`.
Please make sure files are copied with the right ownerships and we do not connect as root.

e.g. using `-a` can be a problem as rsync tends to map users. I think we should really login as jenkins user and avoid `-a` (I see no good reason to use `-W` either) so the outcome is more deterministic

Comment (alxgomz) [reply .github/workflows/deploy-nuxeo.com-benchmarks.yml]: Also avoid using `www` as it relies either on ssh_confg or dns configuration that may not be present.

File: .github/workflows/deploy-nuxeo.com-benchmarks.yml [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Test Nginx Configuration
+        run: ssh root@www "nginx -t"
+
+      - name: Reload Nginx
+        run: ssh root@www "service nginx reload"
```
Comment (alxgomz) [inline .github/workflows/deploy-nuxeo.com-benchmarks.yml]: After checking the content of the existing site, we're simply copying static file. We're not modifying any config oand the site do not embed any either.
We should not need to restart the service at all.

**Review by alxgomz — CHANGES_REQUESTED** [review]
I have confirmed with @ataillefer that there is no pre-prod server for benchmarks.

**Review by alxgomz — APPROVED** [review]

**Pull request merged**